packages/itwinui-react/src/utils/components/OverflowContainer.tsx

```diff
@@ -0,0 +1,141 @@
+} & (
+  | {
+      children: React.ReactNode[];
+      /**
+       * What is rendered at `overflowLocation` when `OverflowContainer` starts overflowing.
+       *
+       * Required if `children: React.ReactNode[]`.
+       */
+      overflowTag: (visibleCount: number) => React.ReactNode;
+      /**
+       * Where the overflowTag is placed. Values:
+       * - start: At the start
+       * - end: At the end
+       * @default 'end'
+       */
+      overflowLocation?: 'start' | 'end';
+    }
+  | {
+      children: (visibleCount: number) => React.ReactNode;
+      overflowTag?: undefined;
+      overflowLocation?: undefined;
+    }
+);
+
+/**
+ * Renders fewer children + an `overflowTag` when it starts overflowing. When not overflowing, it renders all children.
+ * This component listens to resize events and updates the rendered content accordingly.
+ *
+ * Two forms of usage:
+ * 1. `children: React.ReactNode[]`: Pass all the children and an `overflowTag` and this component handles when to show
+ * what, depending on whether the component is overflowing.
+ * 2. `children: (visibleCount: number) => React.ReactNode`: For more customization, pass a function to get the
+ * `visibleCount` and then render custom content based on that.
+ *
+ * @example
+ * <OverflowContainer
+ *   as={MyComponent}
+ *   overflowTag={(visibleCount) => (
+ *     <Text>+${tags.length - (visibleCount - 1)} item(s)</Text> // -1 to account for the overflowTag
+ *   )}
+ *   overflowLocation='start'
+ * >
+ *   {items}
+ * </OverflowContainer>
+ *
+ * @example
+ * <OverflowContainer
+ *   as={MyComponent}
+ *   itemsLength={text.length}
+ * >
+ *   {(visibleCount) => {
+ *     // Custom content dependent on visibleCount
+ *     return (
+ *       <>
+ *         itemsLeft(visibleCount)
+ *         overflowButton(visibleCount)
+ *         itemsRight(visibleCount)
+ *       </>
+ *     );
+ *   }
+ * </OverflowContainer>
+ */
+export const OverflowContainer = React.forwardRef((props, ref) => {
+  const {
+    overflowTag,
+    overflowLocation = 'end',
+    items,
+    children,
+    overflowDisabled = false,
+    overflowOrientation,
+    ...rest
+  } = props;
+
+  const [containerRef, visibleCount] = useOverflow(
+    items,
+    overflowDisabled,
+    overflowOrientation,
+  );
+
+  /**
+   * - `visibleCount === children.length` means that we show all children and no overflow tag.
+   * - `visibleCount < children.length` means that we show visibleCount - 1 children and 1 overflow tag.
+   */
+  const visibleItems = React.useMemo(() => {
+    // Consumer wants complete control over what items are rendered.
+    if (typeof children === 'function' || overflowTag == null) {
+      return null;
+    }
+
+    if (visibleCount >= children.length) {
+      return children;
+    }
+
+    if (overflowLocation === 'start') {
+      return (
+        <>
+          {overflowTag(visibleCount)}
+          {children.slice(children.length - (visibleCount - 1))}
```

Comment:
Whenever doing array operations on children (`.slice`, `.length`), it is recommended to use [`React.Children.toArray`](https://react.dev/reference/react/Children#children-toarray) so that empty and conditional nodes are automatically filtered out

Reply:
I believe this code was moved out of `OverflowContainer` so the change was undone.

For example in breadcrumbs, we currently only rename the `children` prop `items`. We should instead use `React.Children.toArray` (even if we weren't previously using it). 

```jsx
const items = React.Children.toArray(children);
```